pkg/reconciler/kubernetes/kubeclient/kubeclient.go

```diff
@@ -37,6 +37,19 @@ type Metadata struct {
 	Kind      string
 }
 
+//go:generate mockery --name Client
+type Client interface {
```

Comment:
This interface seems wrong to me 🤔 
Now we have 2 overlapping interfaces on essentially the same package (this and https://github.com/kyma-incubator/reconciler/blob/455f43a4dec7df24f98d6932a8bc12288da45730/pkg/reconciler/kubernetes/client.go).

If there are more needs from the k8s client we have for the reconciler we should reconsider the design of this package and of the existing interface, but having 2 interfaces that have the same name, almost identical import path and overlap in functions will lead to confusion, misuse and bugs.

Reply:
The main idea behind this interface was the mocking. In order to test our implementation, we need to mock the pkg/reconciler/kubernetes/kubeclient/kubeclient.go. The mockery tool which we use for mocking purposes requires an interface for that. And in general why not have an interface for the kubeclient, perhaps we will need to have another implementation of that in the future. 
As I see from the original source of the code https://github.com/billiford/go-clouddriver/blob/master/pkg/kubernetes/client.go, the introduced an interface too. 

About the confusing naming I have to agree. But the we already got two clients in the same package anyway. My suggestion would be to just rename the interface in order to avoid the misuses and misinterpretations. What would be your naming suggestion ? I could only come up with Client or ClientStruct until we come we some design changes in this package.

Reply:
As you mention, the kubernetes package already has an interface defining the way to access it in `/pkg/kubernetes/client.go`.
If you need more functionality from that, you can either add it there or use its `Clientset()` function to get the underlying kubernetes full Clientset.

As for the interface only for mocking, you can define that in your own package test file and generate the mock from it. Golang has implicit interfaces, so you do not need to create your mocking interface in `/pkg/kubernetes/kubeclient/`.